drivers/counter/counter_cc23x0_rtc.c

```diff
@@ -0,0 +1,243 @@
+				    const struct counter_alarm_cfg *alarm_cfg)
+{
+	const struct counter_cc23x0_config *config = dev->config;
+	struct counter_cc23x0_data *data = dev->data;
+
+	/* RTC have resolutiuon of 8us */
+	if (counter_ticks_to_us(dev, alarm_cfg->ticks) <= 8) {
+		return -ENOTSUP;
+	}
+
+	uint32_t now = HWREG(config->base + RTC_O_TIME8U);
+
+	/* Calculate next alarm relative to current time in us */
+	uint32_t next_alarm = now + (counter_ticks_to_us(dev, alarm_cfg->ticks) / 8);
+
+	HWREG(config->base + RTC_O_CH0CC8U) = next_alarm;
+	HWREG(config->base + RTC_O_IMASK) = 0x1;
+	HWREG(config->base + RTC_O_ARMSET) = 0x1;
+
+	HWREG(EVTSVT_BASE + EVTSVT_O_CPUIRQ3SEL) = EVTSVT_CPUIRQ16SEL_PUBID_AON_RTC_COMB;
+
+	IRQ_CONNECT(DT_INST_IRQN(0), DT_INST_IRQ(0, priority), counter_cc23x0_isr,
+		    DEVICE_DT_INST_GET(0), 0);
+
+	irq_enable(DT_INST_IRQN(0));
+
+	data->alarm_cfg0.flags = 0;
+	data->alarm_cfg0.ticks = alarm_cfg->ticks;
+	data->alarm_cfg0.callback = alarm_cfg->callback;
+	data->alarm_cfg0.user_data = alarm_cfg->user_data;
+
+	return 0;
+}
+
+static int counter_cc23x0_cancel_alarm(const struct device *dev, uint8_t chan_id)
+{
+	const struct counter_cc23x0_config *config = dev->config;
+
+	/* Unset interrupt source */
+	HWREG(EVTSVT_BASE + EVTSVT_O_CPUIRQ3SEL) = 0x0;
+
+	/* Unarm both channels */
+	HWREG(config->base + RTC_O_ARMCLR) = 0x3;
+
+	return 0;
+}
+
+static int counter_cc23x0_set_top_value(const struct device *dev, const struct counter_top_cfg *cfg)
+{
+	return -ENOTSUP;
+}
+
+static uint32_t counter_cc23x0_get_pending_int(const struct device *dev)
+{
+	const struct counter_cc23x0_config *config = dev->config;
+	struct counter_cc23x0_data *data = dev->data;
+
+	/* Check interrupt and mask */
+	if (HWREG(config->base + RTC_O_RIS) & HWREG(config->base + RTC_O_MIS)) {
+		/* Clear RTC interrupt regs */
+		HWREG(config->base + RTC_O_ICLR) = 0x3;
+		HWREG(config->base + RTC_O_IMCLR) = 0x3;
+
+		uint32_t now = HWREG(config->base + RTC_O_TIME8U);
+
+		if (data->alarm_cfg0.callback) {
+			data->alarm_cfg0.callback(dev, 0, now, data->alarm_cfg0.user_data);
+		}
+
+		return 0;
+	}
+
+	return -ESRCH;
+}
+
+static uint32_t counter_cc23x0_get_top_value(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return -ENOTSUP;
+}
+
+static uint32_t counter_cc23x0_get_freq(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	/*
+	 * From TRM clock for RTC is 24Mhz handled internally
+	 * which is 1/2 from main 48Mhz clock = 24Mhz
+	 * Accessible for user resolution is 8us per bit
+	 * TIME8U [34:3] ~ 9.5h
+	 */
+
+	return (DT_PROP(DT_PATH(cpus, cpu_0), clock_frequency) / 2);
+}
+
+static int counter_cc23x0_start(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	/* RTC timer runs after power-on reset */
+
+	return 0;
+}
+
+static int counter_cc23x0_stop(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	/* Any reset/sleep mode, except for POR, will not stop or reset the RTC timer */
+
+	return 0;
```

Comment:
Since stopping the RTC is not supported, shouldn't the function return `-ENOTSUP` as described in the documentation of [counter_stop()](https://docs.zephyrproject.org/apidoc/latest/group__counter__interface.html#gafaa8198ccff5ffc0491a1424d090c82d)
```suggestion
	return -ENOTSUP;
```